Connection.py

```diff
@@ -0,0 +1,14 @@
+class Connection:
+    def __init__(self, speed_limit, length_miles, entrance_name, exit_name, cars_on_road):
+        self.speed_limit = speed_limit
+        self.length_miles = length_miles
+        self.entrance_name = entrance_name
+        self.exit_name = exit_name
+        self.cars_on_road = cars_on_road
+        if not self.checkInvariants(self):
```

Comment:
generally speaking, you want to avoid having an __init__ function that can fail/raise an Error. Instead call some setup function later that can fail.

Failing in an __init__ is hard to debug and read.